Whisky/Views/Bottle/Pins/PinCreationView.swift

```diff
@@ -0,0 +1,116 @@
+//  either version 3 of the License, or (at your option) any later version.
+//
+//  Whisky is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+//  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+//  See the GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License along with Whisky.
+//  If not, see https://www.gnu.org/licenses/.
+//
+
+import SwiftUI
+import UniformTypeIdentifiers
+import WhiskyKit
+
+struct PinCreationView: View {
+    let bottle: Bottle
+    @State private var newPinName: String = ""
+    @State private var newPinURL: URL?
+    @State private var isValid: Bool = false
+    @State private var isDuplicate: Bool = false
+    @Environment(\.dismiss) private var dismiss
+
+    var body: some View {
+        VStack(alignment: .leading) {
+            Text("pin.title")
+                .bold()
+            Divider()
+            HStack {
+                Text("pin.name")
+                    .frame(maxWidth: .infinity, alignment: .leading)
+                TextField(String(), text: $newPinName)
+                    .frame(width: 180)
+                    .textFieldStyle(.roundedBorder)
+            }
+            HStack {
+                Text("pin.path")
+                    .frame(maxWidth: .infinity, alignment: .leading)
+                Button("create.browse") {
+                    let panel = NSOpenPanel()
+                    panel.canChooseFiles = true
+                    panel.allowedContentTypes = [UTType.exe,
+                                                 UTType(exportedAs: "com.microsoft.msi-installer"),
+                                                 UTType(exportedAs: "com.microsoft.bat")]
+                    panel.directoryURL = newPinURL ?? bottle.url.appending(path: "drive_c")
+                    panel.canChooseDirectories = false
+                    panel.allowsMultipleSelection = false
+                    panel.canCreateDirectories = false
+                    panel.begin { result in
+                        guard result == .OK, let url = panel.urls.first else { return }
+                        let pattern = "\\.(exe|msi|bat)$"
+                        let oldDefaultName = (newPinURL ?? url).lastPathComponent
+                            .replacingOccurrences(of: pattern, with: "", options: .regularExpression)
+                        newPinURL = url
+                        // Only reset newPinName if the textbox hasn't been modified
+                        if newPinName.isEmpty || newPinName == oldDefaultName {
+                            newPinName = url.lastPathComponent
+                                .replacingOccurrences(of: pattern, with: "", options: .regularExpression)
+                        }
+                    }
+                }
+            }
+            if let newPinURL {
```

Comment:
This layout doesn't match what we have in `BottleCreationView`. I'm not to bothered about UI being perfect in the UI as I'll probably tweak things after merge anyway.

Reply:
Fair enough. FWIW, I started with `BottleCreationView` as a base pattern and then diverged predominantly as a result of reviews on this PR. If you have a preferred pattern, I'm happy to revert back to it, or you can adjust as necessary after the fact.